ios/gamma/split-view/RNSSplitViewHostController.swift

```diff
@@ -60,6 +68,18 @@ public class RNSSplitViewHostController: UISplitViewController, ReactMountingTra
     needsSecondaryScreenNavBarAppearanceUpdate = true
   }
 
+  @objc
+  public func setNeedsDisplayModeUpdate() {
+    needsDisplayModeUpdate = true
+  }
+
+  @objc
+  public func consumeDisplayModeUpdateIfNeeded() -> Bool {
+    guard needsDisplayModeUpdate else { return false }
+    needsDisplayModeUpdate = false
+    return true
```

Comment:
It's functionally the same, but IMO much more streamlined to read:

```suggestion
    if needsDisplayMode {
        needsDisplayMode = false
        return true
    }
    return false
```

IIRC `guard` should serve to ensure that some runtime preconditions are satisfied to run the business logic that follows, while this is a very simple branching logic.

Reply:
https://github.com/software-mansion/react-native-screens/pull/3051/commits/71a29f3afa60a87b0cd399845a6e317e56b7cb1b